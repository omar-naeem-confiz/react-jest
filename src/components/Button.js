import React from 'react';
import { Text } from '.';

export const Button = (props) => {
  const { label, onClick, style } = props;
  return (
    <button
      style={{
        backgroundColor: '#017acd',
        height: 54,
        minWidth: 127,
        padding: 10,
        paddingRight: 40,
        paddingLeft: 40,
        borderRadius: 6,
        ...style
      }}
      onClick={onClick}
    >
      <Text
        style={{
          color: 'white',
          fontSize: 16,
          fontWeight: 'bold'
        }}
      >
        {label}
      </Text>
    </button>
  )
}
