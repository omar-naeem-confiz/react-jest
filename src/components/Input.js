import React from 'react';
import { View, Text } from '.';

export const Input = (props) => {
  const { label, value, onChange, style } = props;
  const { width, ...remainingStyle } = style;
  return (
    <View
      style={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
        ...remainingStyle
      }}
    >
      <Text
        style={{
          fontSize: 14,
          paddingBottom: 4
        }}
      >
        {label}
      </Text>
      <input
        style={{
          backgroundColor: 'white',
          height: 48,
          paddingRight: 16,
          paddingLeft: 16,
          fontSize: 24,
          borderRadius: 4,
          border: 'solid 2px #9d9fa2',
          width
        }}
        onChange={(e) => onChange(e.target.value)}
        value={value}
      />
    </View>
  )
}
