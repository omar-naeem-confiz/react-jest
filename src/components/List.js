import React from 'react';
import { View, Text } from '.';

export const List = (props) => {
  const { children, style } = props;
  return (
    <View
      style={{
        flexDirection: 'column',
        margin: 20,
        marginTop: 0,
        ...style
      }}
    >
      {children}
    </View>
  )
}
