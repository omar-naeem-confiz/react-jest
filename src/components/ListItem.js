import React from 'react';
import { View, Text, Button } from '.';

export const ListItem = (props) => {
  const { children, style, onCancel } = props;
  return (
    <View
      style={{
        padding: 20,
        boxShadow: '0px 0px 3px 3px rgba(0,0,0,0.2)',
        borderRadius: 4,
        margin: 20,
        marginTop: 10,
        marginBottom: 10,
        flexDirection: 'row',
        alignItems: 'center',
        zIndex: 1,
        justifyContent: 'space-between',
        ...style
      }}
    >
      <Text>
        {children}
      </Text>
      <Text
        style={{
          cursor: 'pointer',
          marginLeft: 20
        }}
        onClick={onCancel}
      >
        Ã—
      </Text>
    </View>
  )
}
