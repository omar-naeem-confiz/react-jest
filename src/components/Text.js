import React from 'react';
import { View } from '.';

export const Text = (props) => {
  const { children, style, ...rest } = props;
  return (
    <View
      style={{
        color: 'rgba(0,0,0,0.87)',
        fontSize: 16,
        ...style
      }}
      {...rest}
    >
      {children}
    </View>
  )
}
