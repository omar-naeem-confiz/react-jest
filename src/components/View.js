import React from 'react';

export const View = (props) => {
  const { style, children, ...rest } = props;
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        ...style
      }}
      {...rest}
    >
      {children}
    </div>
  )
}
