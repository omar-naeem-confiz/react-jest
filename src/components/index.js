export * from './View';
export * from './Button';
export * from './Input';
export * from './Text';
