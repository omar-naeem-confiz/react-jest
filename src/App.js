import React, { useState, useEffect } from 'react';
import { View } from './components'
import { Todos } from './pages';
import { getList, addItem, removeItem } from './apis';

function App() {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);

  const onAdd = async (value) => {
    setLoading(true);
    const response = await addItem(value);
    setData(response.data);
    setLoading(false);
  }
  const onRemove = async (index) => {
    setLoading(true);
    const response = await removeItem(index);
    setData(response.data);
    setLoading(false);
  }
  useEffect(async () => {
    setLoading(true);
    const response = await getList();
    setData(response.data);
    setLoading(false);
  }, [])
  return (
    <View
      className="App"
    >
      <Todos
        data={data}
        onAdd={onAdd}
        onRemove={onRemove}
        loading={loading}
      />
    </View>
  );
}

export default App;
