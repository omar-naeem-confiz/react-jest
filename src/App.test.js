import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import { ListItem } from './components';

jest.mock('./apis/todos.js');

beforeEach(() => {
  localStorage.clear();
  localStorage.setItem('list', "[\"item1\", \"item2\"]");
});

it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('mount whole component with mocks', () => {
  const wrapper = mount(<App />);
  // expect(wrapper.find(ListItem)).toHaveLength(2);
  expect(wrapper).toMatchSnapshot();
});
