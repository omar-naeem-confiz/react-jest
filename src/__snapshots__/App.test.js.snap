// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount whole component with mocks 1`] = `
<App>
  <View
    className="App"
  >
    <div
      className="App"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Todos
        data={Array []}
        loading={true}
        onAdd={[Function]}
        onRemove={[Function]}
      >
        <View
          className="App"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.1)",
              "height": "100vh",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="App"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0.1)",
                "display": "flex",
                "flexDirection": "column",
                "height": "100vh",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 6,
                  "boxShadow": "0px 0px 3px 3px rgba(0,0,0,0.1)",
                  "height": 600,
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 6,
                    "boxShadow": "0px 0px 3px 3px rgba(0,0,0,0.1)",
                    "display": "flex",
                    "flexDirection": "column",
                    "height": 600,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                      "padding": 20,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "display": "flex",
                        "flexDirection": "row",
                        "padding": 20,
                      }
                    }
                  >
                    <Input
                      label="Todo"
                      onChange={[Function]}
                      style={
                        Object {
                          "width": 400,
                        }
                      }
                      value=""
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "display": "flex",
                            "flexDirection": "column",
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "display": "flex",
                              "flexDirection": "column",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "fontSize": 14,
                                "paddingBottom": 4,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "color": "rgba(0,0,0,0.87)",
                                  "fontSize": 14,
                                  "paddingBottom": 4,
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "color": "rgba(0,0,0,0.87)",
                                    "display": "flex",
                                    "flexDirection": "column",
                                    "fontSize": 14,
                                    "paddingBottom": 4,
                                  }
                                }
                              >
                                Todo
                              </div>
                            </View>
                          </Text>
                          <input
                            onChange={[Function]}
                            style={
                              Object {
                                "backgroundColor": "white",
                                "border": "solid 2px #9d9fa2",
                                "borderRadius": 4,
                                "fontSize": 24,
                                "height": 48,
                                "paddingLeft": 16,
                                "paddingRight": 16,
                                "width": 400,
                              }
                            }
                            value=""
                          />
                        </div>
                      </View>
                    </Input>
                    <Button
                      label="Add"
                      onClick={[Function]}
                      style={
                        Object {
                          "marginLeft": 20,
                          "width": 100,
                        }
                      }
                    >
                      <button
                        onClick={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#017acd",
                            "borderRadius": 6,
                            "height": 54,
                            "marginLeft": 20,
                            "minWidth": 127,
                            "padding": 10,
                            "paddingLeft": 40,
                            "paddingRight": 40,
                            "width": 100,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "color": "white",
                                "fontSize": 16,
                                "fontWeight": "bold",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "color": "white",
                                  "display": "flex",
                                  "flexDirection": "column",
                                  "fontSize": 16,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              Add
                            </div>
                          </View>
                        </Text>
                      </button>
                    </Button>
                  </div>
                </View>
                <List
                  style={
                    Object {
                      "overflowY": "auto",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "column",
                        "margin": 20,
                        "marginTop": 0,
                        "overflowY": "auto",
                        "width": "100%",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flexDirection": "column",
                          "margin": 20,
                          "marginTop": 0,
                          "overflowY": "auto",
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "fontSize": 18,
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "color": "rgba(0,0,0,0.87)",
                              "fontSize": 18,
                              "padding": 10,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "rgba(0,0,0,0.87)",
                                "display": "flex",
                                "flexDirection": "column",
                                "fontSize": 18,
                                "padding": 10,
                              }
                            }
                          >
                            Loading...
                          </div>
                        </View>
                      </Text>
                    </div>
                  </View>
                </List>
              </div>
            </View>
          </div>
        </View>
      </Todos>
    </div>
  </View>
</App>
`;

exports[`renders without crashing 1`] = `
<View
  className="App"
>
  <Todos
    data={Array []}
    loading={false}
    onAdd={[Function]}
    onRemove={[Function]}
  />
</View>
`;
